docs: add onboarding maintainers guide (#1638)
asyncapi/community: docs/onboarding-guide/Onboarding-maintainer-guide.md

# Onboarding Guide for AsyncAPI Maintainers

Welcome to the AsyncAPI Maintainer Onboarding Guide! 

> This document provides you with guidelines about everything you need to know to begin your journey as a maintainer within the AsyncAPI ecosystem. Maintainers are the backbone of any open-source project, helping with different activities that help the project to stay on track and foster a healthy, productive community.

Reviewer: ```suggestion
This document provides you with guidelines about everything you need to know to begin your journey as a maintainer within the AsyncAPI ecosystem. Maintainers are the backbone of any open-source project, helping with different activities that help the project stay on track and foster a healthy, productive community.
```

Before you can become a **maintainer**, you need to start as a **contributor**. The journey from contributor to maintainer is a rewarding one, and it involves the following steps:

### 1. **Pick an Issue**

- **Join existing PR reviews**: If you're not sure where to start, begin by reviewing open [PR](https://github.com/pulls?q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+archived%3Afalse+) within the organization. This will give you a high-level understanding of the projects and where your contributions might fit in.

- **Look for "*good first issue*" labels**: These [issues](https://github.com/issues?page=1&q=is%3Aopen+org%3Aasyncapi+sort%3Aupdated-desc+label%3A%22good+first+issue%22) are beginner-friendly and will help you get familiar with the project’s structure. Additionally, you can check out the [#97_bot-github-new-issues-prs](https://asyncapi.slack.com/archives/C01J06RL10X) channel on Slack for new issues and PRs.

- **Participate in live streams**: AsyncAPI maintainers sometimes host [live streams](https://www.asyncapi.com/community/events) where they walk through parts of the project. You can request a session on the specific area you want to contribute to.

> [!NOTE]

> > Make sure whatever issue you pick isn't marked "Do-not-merge" else your PR won't be merged.

Reviewer: ```suggestion
> Make sure whatever issue you pick isn't marked "Do-not-merge," or else your PR won't be merged.
```

# Onboarding Guide for AsyncAPI Maintainers

Welcome to the AsyncAPI Maintainer Onboarding Guide! 

> This document provides you with guidelines about everything you need to know to begin your journey as a maintainer within the AsyncAPI ecosystem. Maintainers are the backbone of any open-source project, helping with different activities that help the project stay on track and foster a healthy, productive community.

Reviewer 2: ```suggestion
This document aims to provide comprehensive guidelines about everything you need to know to begin your journey as a maintainer within the AsyncAPI ecosystem. Maintainers are the backbone of any open-source project, helping with different activities that help the project stay on track and foster a healthy, productive community.
```

Before we go into getting you started, let's try and clarify just *who* a maintainer is.

## Who is an AsyncAPI Maintainer?

A maintainer is an individual who plays a crucial role in overseeing and guiding the development and growth of an open-source project. As a maintainer for AsyncAPI, you'll be responsible for:

- Overseeing the technical direction of the project
- Helping with reviewing and resolving issues and pull requests (PR)
- Managing workflows and GitHub Actions to automate tasks

> - Enforcing coding standards

Reviewer 2: ```suggestion
- Enforcing coding standards
- Enforcing relevant and up-to-date documentation
```

   - **Make changes**: Implement the changes required to resolve the issue you picked. Ensure your code adheres to the project’s coding standards.

   - **Submit a Pull Request**: Once you’re done, submit a pull request to the main repository. Make sure to include a detailed description of your changes.

   - **Participate in discussions**: Engage with maintainers and other contributors in the PR comments section. This will help you understand the project better and improve your contributions.

### 3. **Get your Pull Request Merged**

   - After submitting a pull request, it’s time for the maintainers to review it.

   - If you notice no one reviewing your PR (which is rare) or if you need a review urgently, don’t hesitate to ping the maintainers on Slack or GitHub

   - Make sure all the checks (tests, style checks, etc.) turn green for a smooth merge.

### 4. **Receive an invitation to become a Maintainer 🎉**

   - **Recognition**: After contributing consistently, the maintainers will invite you to join the team. This invitation is a recognition of your contributions and dedication.

>    - If you feel you've made enough contributions, but haven't received an invitation, you can open an issue in the relevant repository to discuss your contributions with the maintainers. You can see an example of such an issue [here](https://github.com/asyncapi/cli/issues/1616).

Reviewer 3: > If you feel you've made **enough** contributions

How many contributions are considered "enough"? I would rather rephrase it to something like that to get rid of ambiguity:

If you haven't received an invitation despite contributing consistently, you can open an issue in the corresponding repository to discuss your contributions with the maintainers. Check an [example of such an issue](https://github.com/asyncapi/cli/issues/1616).

> ## Some Useful Links
> 
> - [AsyncAPI Contributor’s Guide](https://github.com/asyncapi/cli/blob/master/CONTRIBUTING.md): Learn how to set up, contribute, and work on pull requests.
> - [GitHub Workflow Setup Guide](https://github.com/asyncapi/community/blob/master/git-workflow.md): A detailed walkthrough of setting up GitHub Actions and managing workflows.
> - [AsyncAPI Slack Community](https://www.asyncapi.com/slack-invite): Join the community and interact with fellow contributors and maintainers.

Reviewer 3: I would consider deleting this section as links 1 and 3 are available in the community home page, and the link 2 is available in the section 2: **Open a Pull Request**. But here, it is misleading, it's not about GitHub Actions.

```suggestion
```

Before we go into getting you started, let's try and clarify just *who* a maintainer is.

## Who is an AsyncAPI Maintainer?

A maintainer is an individual who plays a crucial role in overseeing and guiding the development and growth of an open-source project. As a maintainer for AsyncAPI, you'll be responsible for:

- Overseeing the technical direction of the project
- Helping with reviewing and resolving issues and pull requests
- Managing workflows and GitHub Actions to automate tasks
- Enforcing coding standards
- Identifying and appointing new maintainers
- Mentoring new contributors and helping them navigate their journey
- Recognizing and rewarding contributions to foster community engagement

Essentially, you serve as a person that *binds* the project together and guarantees that everything runs smoothly.

## Steps to becoming a Maintainer

Before you can become a **maintainer**, you need to start as a **contributor**. The journey from contributor to maintainer is a rewarding one, and it involves the following steps:

### 1. **Pick an Issue**

> - **Join existing PR reviews**: If you're not sure where to start, begin by reviewing open [pull requests](https://github.com/asyncapi/community/pulls). This will give you an understanding of the code and where your contributions might fit in.

Reviewer 3: Please note the updated link I mentioned in my comment. The current link points to pull requests within the community repository only. As you're talking about "This will give you an understanding of the **code**", the community repository contains only documentation in Markdown and no code. My updated link points to **all the repos within the AsyncAPI organization** that include pull requests related to code as well.

> - **Look for "*good first issue*" labels**: These [issues](https://github.com/asyncapi/community/issues) are beginner-friendly and will help you get familiar with the project’s structure. Additionally, you can check out the [#97_bot-github-new-issues-prs](https://asyncapi.slack.com/archives/C01J06RL10X) channel on Slack for new issues and PRs.

Reviewer 3: The same with the issues link

- **Participate in live streams**: AsyncAPI maintainers sometimes host [live streams](https://www.asyncapi.com/community/events) where they walk through parts of the project. You can request a session on the specific area you want to contribute to.

> > **NOTE**: Make sure whatever issue you pick isn't marked "Do-not-merge" else your PR won't be merged.

Reviewer 3: Please note the updated formatting for the note in my comment.

### 2. **Open a Pull Request**

> For a comprehensive guide on how to open a pull request, refer to the [AsyncAPI Git Workflow Guide](https://github.com/asyncapi/community/blob/master/git-workflow.md).

Reviewer 3: Missed suggestion

- **Fork the repository**: Fork the repository you want to contribute to and create a new branch for your changes.

- **Make changes**: Implement the changes required to resolve the issue you picked. Ensure your code adheres to the project’s coding standards.

- **Submit a Pull Request**: Once you’re done, submit a pull request to the main repository. Make sure to include a detailed description of your changes.

> - **Participate in discussions**: Engage with maintainers and other contributors in the PR comments section. This will help you understand the project better and improve your contributions.

Reviewer 3: Missed suggestions here as well.

### 3. **Get your Pull Request Merged**

- After submitting a pull request, it’s time for the maintainers to review it.

- If you notice no one reviewing your PR (which is rare) or if you need a review urgently, don’t hesitate to ping the maintainers on Slack or GitHub

> - Make sure all the checks (tests, style checks, etc.) turn green for a smooth merge.

Reviewer 3: Missed suggestions

### 4. **Receive an invitation to become a Maintainer 🎉**

- **Recognition**: After contributing consistently, the maintainers will invite you to join the team. This invitation is a recognition of your contributions and dedication.

> - If you haven't received an invitation despite contributing consistently, you can open an issue in the corresponding repository to discuss your contributions with the maintainers. You can see an example of such an issue [here](https://github.com/asyncapi/cli/issues/1616). 

Reviewer 3: Missed suggestions
